gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/v4/subscription/DefaultSubscriptionDispatcher.java

```diff
@@ -50,21 +63,37 @@ public class DefaultSubscriptionDispatcher extends AbstractService<SubscriptionD
 
     public static final int ON_SUBSCRIPTION_ERROR_RETRY_COUNT = 5;
     public static final int ON_SUBSCRIPTION_ERROR_RETRY_DELAY_MS = 3_000;
+
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultSubscriptionDispatcher.class);
     private static final String SUBSCRIPTION_ENTRYPOINT_FIELD = "entrypointId";
+
+    protected static final String TRACING_SPAN_NAME = "SUBSCRIPTION";
+    protected static final String ATTR_INTERNAL_TRACING_SPAN = "subscription-tracing-span";
+
     private final Map<String, Subscription> activeSubscriptions = new ConcurrentHashMap<>();
     private final Map<String, Disposable> activeDisposables = new ConcurrentHashMap<>();
 
     private final SubscriptionAcceptorResolver subscriptionAcceptorResolver;
     private final ObjectMapper mapper = new ObjectMapper();
     private final SubscriptionExecutionRequestFactory subscriptionExecutionRequestFactory;
+    private final SubscriptionPlatformProcessorChainFactory platformProcessorChainFactory;
+    private final Vertx vertx;
+    private final List<ChainHook> processorChainHooks;
+    protected boolean tracingEnabled;
 
     public DefaultSubscriptionDispatcher(
         SubscriptionAcceptorResolver subscriptionAcceptorResolver,
-        SubscriptionExecutionRequestFactory subscriptionExecutionRequestFactory
+        SubscriptionExecutionRequestFactory subscriptionExecutionRequestFactory,
+        SubscriptionPlatformProcessorChainFactory platformProcessorChainFactory,
+        boolean tracingEnabled,
+        Vertx vertx
     ) {
         this.subscriptionAcceptorResolver = subscriptionAcceptorResolver;
         this.subscriptionExecutionRequestFactory = subscriptionExecutionRequestFactory;
+        this.platformProcessorChainFactory = platformProcessorChainFactory;
+        this.tracingEnabled = tracingEnabled;
+        this.processorChainHooks = tracingEnabled ? List.of(new TracingHook("processor-chain")) : new ArrayList<>();
```

Comment:
I am not sure to understand properly; Is there a typo in the hook's name?

Reply:
The id of the hook is `hook-tracing`, defined in TracingHook.
`processor-chain` is the key of the TracingHook, that will be used to create span names like, for example `processor-chain-before-api-handle` (as we can see in screenshot).
Same key than DefaultHttpRequestDispatcher tracingHook.
Sound relevant to you ?